File: bucles.cpp
#include <iostream>
#include <chrono>
#include <cstdlib>
using namespace std;
using namespace std::chrono;

const int MAX = 10000;

double A[MAX][MAX], x[MAX], y[MAX];

void iniciar() {
    srand(42);
    for(int i = 0; i < MAX; i++){
        for(int j = 0; j < MAX; j++) A[i][j] = rand() % 100;
        x[i] = i;
        y[i] = 0;
    }
}

void first_pair() {
    for(int i = 0; i < MAX; i++){
        for(int j = 0; j < MAX; j++) y[i] += A[i][j] * x[j];
    }
}

void second_pair() {
    for(int j = 0; j < MAX; j++){
        for(int i = 0; i < MAX; i++) y[i] += A[i][j] * x[j];
    }
}

int main() {
    iniciar();
    auto start = high_resolution_clock::now();
    first_pair();
    auto stop = high_resolution_clock::now();
    auto duration = duration_cast<milliseconds>(stop - start);
    cout << "Tiempo first pair: " << duration.count() << "ms" << endl;
    iniciar();
    start = high_resolution_clock::now();
    second_pair();
    stop = high_resolution_clock::now();
    duration = duration_cast<milliseconds>(stop - start);
    cout << "Tiempo second pair: " << duration.count() << "ms" << endl;
}

File: bucles.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

const MAX = 10000

var A [MAX][MAX]float64
var x, y [MAX]float64

func iniciar() {
	rand.Seed(42)
	for i := 0; i < MAX; i++ {
		for j := 0; j < MAX; j++ {
			A[i][j] = float64(rand.Intn(100))
		}
		x[i] = float64(i)
		y[i] = 0
	}
}

func first_pair() {
	for i := 0; i < MAX; i++ {
		for j := 0; j < MAX; j++ {
			y[i] += A[i][j] * x[j]
		}
	}
}

func second_pair() {
	for j := 0; j < MAX; j++ {
		for i := 0; i < MAX; i++ {
			y[i] += A[i][j] * x[j]
		}
	}
}

func main() {
	iniciar()
	start := time.Now()
	first_pair()
	fmt.Printf("Tiempo first pair: %v\n", time.Since(start))

	iniciar()
	start = time.Now()
	second_pair()
	fmt.Printf("Tiempo second pair: %v\n", time.Since(start))
}
